pandas/core/case_when.py

```
@@ -0,0 +1,151 @@
+            "The number of boolean conditions should be equal "
+            "to the number of replacements. "
+            "However, the total number of conditions and replacements "
+            f"is {len(args)}, which is an odd number."
+        )
+
+    conditions = args[-2::-2]
+    replacements = args[-1::-2]
+    common_dtype = [infer_dtype_from(replacement)[0] for replacement in replacements]
+    if default is not None:
+        arg_dtype, _ = infer_dtype_from(default)
+        common_dtype.append(arg_dtype)
+    if len(set(common_dtype)) > 1:
+        common_dtype = find_common_type(common_dtype)
+        replacements = [
+            construct_1d_arraylike_from_scalar(
+                value=replacement, length=len(condition), dtype=common_dtype
+            )
+            if is_scalar(replacement)
+            else replacement.astype(common_dtype, copy=False)
+            if isinstance(replacement, ABCSeries)
+            else pd_array(replacement, dtype=common_dtype, copy=False)
+            for condition, replacement in zip(conditions, replacements)
+        ]
+        if (default is not None) and isinstance(default, ABCSeries):
+            default = default.astype(common_dtype, copy=False)
+    else:
+        common_dtype = common_dtype[0]
+    # TODO: possibly extend this to a DataFrame?
+    if not isinstance(default, ABCSeries):
+        default_index = len(args[0])
+        for condition in conditions[::-1]:
+            if isinstance(condition, ABCSeries):
+                default_index = condition.index
+                break
+        default = Series(default, index=default_index, dtype=common_dtype)
+    counter = len_args // 2 - 1
+    counter = range(counter, -1, -1)
+    for position, condition, replacement in zip(counter, conditions, replacements):
+        try:
+            default = default.mask(
+                condition, other=replacement, axis=0, inplace=False, level=level
+            )
+        except Exception as error:
```

I would remove this try except and have mask raise it's error normally

---

The idea here is to keep track of which `condition, replacement` failed. More like `condition1` failed, this is why it failed. Devolving to mask error directly and you lose the error tracking. I assume the error tracking would be useful to the user.